src/msg/async/AsyncMessenger.h

```diff
@@ -356,10 +356,8 @@ class AsyncMessenger : public SimplePolicyMessenger {
       Mutex::Locker l(deleted_lock);
       if (deleted_conns.erase(existing)) {
         existing->get_perf_counter()->dec(l_msgr_active_connections);
-        conns.erase(it);
-      } else if (conn != existing) {
-        return -1;
       }
+      conns.erase(it);
```

Comment:
oic. so the `existing` is retrieved from `deleted_conn` and `conns`, and is not put back to `deleted_conn` yet. 

probably we can just remove the line above. `conns[conn->peer_addr] = conn;` will take care of this.

@yuyuyu101 what do you think?